### PR TITLE
Add coding conventions

### PR DESCRIPTION
**Pull request by HappyRay**

### CODE REVIEW

**Comment by HappyRay** [comment]
Thanks @juhoautio! 😄

**Comment by chengren311** [comment]
I'm wondering what the convention of static methods are? Should they all be put into an util class?

**Comment by HappyRay** [comment]
> I'm wondering what the convention of static methods are? Should they all be put into an util class?

The "favor small classes " convention applies too. 
There are cases where it's better to keep static methods in the same class e.g. to implement a factory pattern. 

I would also like to keep the list of coding conventions as small as possible to help people remember.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -59,3 +61,46 @@ It allows us to do following when saving a file:
 * Remove unnecessary semicolon
 * Add missing @Override annotations
 
+## Coding conventions
```
Comment (kunkun-tang) [inline CONTRIBUTING.md]: Do we name conventions?

Comment (HappyRay) [reply CONTRIBUTING.md]: It's covered in the referenced Google style guide.

**Comment by HappyRay** [comment]
> In this doc, do we need to state that Azkaban build doesn't allow warning. If we have to bring code with warning, we must suppress it.

Do you mean errorprone warnings? @reallocf has made the change to enforce the no warning rule. So I don't think it is necessary to mention it here.

**Comment by reallocf** [comment]
I'd rather the standard be that we fix warnings instead of suppressing them :wink:

We can mention it if we want to. But it should definitely say that any suppressed warnings must be justified somewhere - in the commit message at a minimum and ideally in a comment within the actual code.

**Pull request merged**